plume-db-querydsl/src/main/java/com/coreoz/plume/db/querydsl/pagination/SQLPaginatedQuery.java

```diff
@@ -0,0 +1,61 @@
+package com.coreoz.plume.db.querydsl.pagination;
+
+import com.querydsl.core.QueryResults;
+import com.querydsl.core.types.OrderSpecifier;
+import com.querydsl.sql.SQLQuery;
+
+public class SQLPaginatedQuery<U> {
+
+    private final SQLQuery<U> sqlQuery;
+
+    private SQLPaginatedQuery(SQLQuery<U> sqlQuery) {
+        this.sqlQuery = sqlQuery;
+    }
+
+    public static <U> SQLPaginatedQuery<U> fromQuery(SQLQuery<U> sqlQuery) {
+        return new SQLPaginatedQuery<>(sqlQuery);
+    }
+
+    public SQLPaginatedQuery<U> withSort(SortOption sortOption) {
+        return new SQLPaginatedQuery<>(
+            sqlQuery
+                .orderBy(
+                    new OrderSpecifier<>(
+                        sortOption.sortDirection(),
+                        sortOption.sortOn().getSortColumn()
+                    )
+                )
+        );
+    }
+
+    public Page<U> paginate(Pageable pageable) {
+        return this.fetchPage(
+            this.sqlQuery,
+            pageable
+        );
+    }
+
+    public Page<U> paginate(Integer size, Integer page) {
+        return this.paginate(Pageable.ofPageSize(size).withPage(page));
+    }
+
+    private Page<U> fetchPage(SQLQuery<U> baseQuery, Pageable pageable) {
```

Comment:
Ce serait bien d'avoir une méthode fetchList qui retourne la page demandée sans retourner les métriques : ça évite une requête SQL

Reply:
Idéalement, il faudrait aussi fournir une implémentation de `fetchSlice` qui évite aussi de faire une requête supplémentaire : https://stackoverflow.com/questions/49918979/page-vs-slice-when-to-use-which